homeassistant/components/websocket_api/connection.py

```diff
@@ -3,48 +3,50 @@
 
 import asyncio
 from collections.abc import Hashable
-from typing import Any, Callable
+from typing import TYPE_CHECKING, Any, Callable
 
 import voluptuous as vol
 
-from homeassistant.core import Context, callback
+from homeassistant.auth.models import RefreshToken, User
+from homeassistant.core import Context, HomeAssistant, callback
 from homeassistant.exceptions import HomeAssistantError, Unauthorized
 
 from . import const, messages
 
-# mypy: allow-untyped-calls, allow-untyped-defs
+if TYPE_CHECKING:
+    from .http import WebSocketAdapter
 
 
 class ActiveConnection:
     """Handle an active websocket client connection."""
 
-    def __init__(self, logger, hass, send_message, user, refresh_token):
+    def __init__(
+        self,
+        logger: WebSocketAdapter,
+        hass: HomeAssistant,
+        send_message: Callable[[str | dict[str, Any]], None],
+        user: User,
+        refresh_token: RefreshToken,
+    ) -> None:
         """Initialize an active connection."""
         self.logger = logger
         self.hass = hass
         self.send_message = send_message
         self.user = user
-        if refresh_token:
-            self.refresh_token_id = refresh_token.id
-        else:
-            self.refresh_token_id = None
-
+        self.refresh_token_id = refresh_token.id
```

Comment:
`ActiveConnection` is created only in `websocket_api/api.py` where it's explicitly checked that `refresh_token_id is not None`.